From 7540320b20cdc2eb277cdf1bfac11cc51b1c1f72 Mon Sep 17 00:00:00 2001
From: Chia-Ping Tsai <chia7712@gmail.com>
Date: Sun, 19 Mar 2017 19:10:24 +0800
Subject: [PATCH 340/362] HBASE-17803 PE always re-creates table when we
 specify the split policy

(cherry picked from commit 02d9bf0c57b447ab0f4fa93a3858bd27c43e0cf9)

Change-Id: I0f0d279b43cdd4ab844c37204ef8d36e48778577
Reason:    Supportability
Author:    Chia-Ping Tsai <chia7712@gmail.com>
Ref:       CDH-53566
---
 .../apache/hadoop/hbase/PerformanceEvaluation.java |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
index a1e0a69..07c8c3a 100644
--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
+++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
@@ -45,6 +45,7 @@ import java.util.concurrent.Future;
 import com.google.common.base.Objects;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 
+import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.conf.Configuration;
@@ -326,7 +327,8 @@ public class PerformanceEvaluation extends Configured implements Tool {
     // recreate the table when user has requested presplit or when existing
     // {RegionSplitPolicy,replica count} does not match requested.
     if ((exists && opts.presplitRegions != DEFAULT_OPTS.presplitRegions)
-      || (!isReadCmd && desc != null && desc.getRegionSplitPolicyClassName() != opts.splitPolicy)
+      || (!isReadCmd && desc != null &&
+          !StringUtils.equals(desc.getRegionSplitPolicyClassName(), opts.splitPolicy))
       || (!isReadCmd && desc != null && desc.getRegionReplication() != opts.replicas)) {
       needsDelete = true;
       // wait, why did it delete my table?!?
-- 
1.7.9.5

